packages/compass-serverstats/src/components/index.tsx

```diff
@@ -0,0 +1,85 @@
+import './index.less';
+
+import React, { useEffect, useRef } from 'react';
+import { Banner, WorkspaceContainer, css, spacing } from '@mongodb-js/compass-components';
+
+import GraphsComponent from './server-stats-graphs-component';
+import { realTimeDispatcher } from '../d3';
+import ListsComponent from './server-stats-lists-component';
+import { DBErrorComponent } from './dberror-component';
+import DBErrorStore from '../stores/dberror-store';
+import ServerStatsStore from '../stores/server-stats-graphs-store';
+import { ServerStatsToolbar } from './server-stats-toolbar';
+import Actions from '../actions';
+
+const REFRESH_STATS_INTERVAL_MS = 1000;
+
+const workspaceContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  overflow: 'auto'
+});
+
+const workspaceBackground = '#3D4247';
+
+const workspaceBackgroundStyles = css({
+  background: workspaceBackground
+});
+
+const workspaceStyles = css({
+  padding: spacing[4],
+  marginBottom: spacing[6],
+  display: 'flex',
+  flexWrap: 'wrap',
+  justifyContent: 'space-around',
+  flexGrow: 1
+});
+
+const mongosWarningStyles = css({
+  margin: spacing[2],
+  marginBottom: 0
+});
+
+/**
+ * Renders the entire performance tab, including charts and lists.
+ */
+function PerformanceComponent() {
+  const eventDispatcher = useRef(realTimeDispatcher());
+
+  useEffect(() => {
+    return () => {
+      // Reset the store on unmount so that when this is remounted
+      // the errors and previous state are reset.
+      Actions.restart();
```

Comment:
This is a change in behavior from before. This makes it so we reset the stores of the `compass-serverstats` plugin when we unmount. This should fix both the issue where the performance page would not display on disconnect and connect, and the issue where the errors would not display if a user leaves the performance tab and goes back to it.  COMPASS-4148 

Places that listen to this action:
https://github.com/mongodb-js/compass/search?q=Actions.restart

Reply:
Do we need some manual cleanup for whatever state was there before restarting too, or that's enough?

Reply:
As part of the startup, the entire store runs this restart command. The state there is already manually cleaned up.
The places that are listening to this `restart` event currently already perform the cleanup. Those places can be found here: https://github.com/mongodb-js/compass/search?q=Actions.restart